### ISSUE TITLE
[QUESTION] Best Practice: Multiple Playwright Instances

### ISSUE THREAD

**Issue by jwiebels** [microsoft/playwright-python#686]
What is the best practice 1. "concurrency-wise" and 2. "resource-wise" to run e.g. a browser automation, which logs into multiple different accounts on my website, performs some clicks, and logs out? (assuming I do not intend to use Docker containers / K8s etc. and only want to run tests on one machine)
Do I have to launch multiple instances of Playwright with unique contexts, create one instance with multiple contexts etc.?

Best regards

**Comment by pavelfeldman** [comment]
- If you'd like to run your tests in parallel, you should launch multiple instances of Playwright, we run them in different OS processes when we parallelize.
- Inside Playwright, for each user story you should create a context, perform actions and close it.

https://github.com/microsoft/playwright-pytest would do context organization and parallelization for you out of the box. It is based on pytest that supports parallel test execution.

**Comment by mxschmitt** [comment]
Closed for now since it seemed answered, feel free to reopen / create a new issue for further questions.
